src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionTests.cs

```diff
@@ -31210,4 +31326,1150 @@ static Expression<Func<Func<string, string>>> Test()
+    {
+        public new string ToString() => "";
+        public new int P { get => 0; }
+        public new int this[int j] { get => 0; }
+    }
+}
+""";
+        var comp = CreateCompilation(source);
+        comp.VerifyEmitDiagnostics(
+            // (5,27): error CS0106: The modifier 'new' is not valid for this item
+            //         public new string ToString() => "";
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "ToString").WithArguments("new").WithLocation(5, 27),
+            // (6,24): error CS0106: The modifier 'new' is not valid for this item
+            //         public new int P { get => 0; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "P").WithArguments("new").WithLocation(6, 24),
+            // (7,24): error CS0106: The modifier 'new' is not valid for this item
+            //         public new int this[int j] { get => 0; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "this").WithArguments("new").WithLocation(7, 24));
+    }
+
+    [Fact]
+    public void Validation_Modifiers_Override()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        public override string ToString() => "";
+        public override int P { get => 0; }
+        public override int this[int j] { get => 0; }
+    }
+}
+""";
+        var comp = CreateCompilation(source);
+        comp.VerifyEmitDiagnostics(
+            // (5,32): error CS0106: The modifier 'override' is not valid for this item
+            //         public override string ToString() => "";
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "ToString").WithArguments("override").WithLocation(5, 32),
+            // (6,29): error CS0106: The modifier 'override' is not valid for this item
+            //         public override int P { get => 0; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "P").WithArguments("override").WithLocation(6, 29),
+            // (7,29): error CS0106: The modifier 'override' is not valid for this item
+            //         public override int this[int j] { get => 0; }
+            Diagnostic(ErrorCode.ERR_BadMemberFlag, "this").WithArguments("override").WithLocation(7, 29));
+    }
+
+    [Fact]
+    public void Validation_Modifiers_Partial()
+    {
+        string source = """
+static class E
+{
+    extension(int i)
+    {
+        partial void M();
+        partial void M() { }
+        partial int P { get; set; }
+        partial int P { get => 0; set { } }
+        partial int this[int j] { get; }
+        partial int this[int j] { get => 0; }
+    }
+}
+""";
+        var comp = CreateCompilation(source);
+        comp.VerifyEmitDiagnostics(
+            // (5,22): error CS0751: A partial member must be declared within a partial type
+            //         partial void M();
+            Diagnostic(ErrorCode.ERR_PartialMemberOnlyInPartialClass, "M").WithLocation(5, 22),
+            // (6,22): error CS0751: A partial member must be declared within a partial type
```

Comment:
>error CS0751: A partial member must be declared within a partial type [](http://example.com/codeflow?start=23&length=69)

Do we understand why this error is reported twice for methods and only once for properties/indexers?

Reply:
This is an existing behavior. There's a difference between properties and methods in that regard.